Source/cmake/BubblewrapSandboxChecks.cmake

```
@@ -1,42 +1,41 @@
+    if (NOT DEFINED BWRAP_EXECUTABLE)
+        if (CMAKE_CROSSCOMPILING)
+            message(FATAL_ERROR "bwrap executable is needed for ENABLE_BUBBLEWRAP_SANDBOX. Unable to autodetect the path when cross-compiling. "
+                                "Please define define the CMake variable BWRAP_EXECUTABLE with the run-time full-path to the 'bwrap' program.")
+        else ()
+            find_program(BWRAP_EXECUTABLE bwrap)
+            if (NOT BWRAP_EXECUTABLE)
+                message(FATAL_ERROR "bwrap executable is needed for ENABLE_BUBBLEWRAP_SANDBOX. "
+                       "Either install it or use the CMake variable BWRAP_EXECUTABLE to define the runtime path.")
+            endif ()
+        endif ()
     endif ()
 
-    if (NOT CMAKE_CROSSCOMPILING)
-        execute_process(
-            COMMAND "${BWRAP_EXECUTABLE}" --version
-            RESULT_VARIABLE BWRAP_RET
-            OUTPUT_VARIABLE BWRAP_OUTPUT
-        )
-        if (BWRAP_RET)
-            message(FATAL_ERROR "Failed to run ${BWRAP_EXECUTABLE}")
-        endif ()
-        string(REGEX MATCH "[0-9]+\\.[0-9]+\\.[0-9]+" BWRAP_VERSION "${BWRAP_OUTPUT}")
-        if (NOT "${BWRAP_VERSION}" VERSION_GREATER_EQUAL "0.3.1")
-            message(FATAL_ERROR "bwrap must be >= 0.3.1 but ${BWRAP_VERSION} found")
+    if (NOT DEFINED DBUS_PROXY_EXECUTABLE)
+        if (CMAKE_CROSSCOMPILING)
+            message(FATAL_ERROR "xdg-dbus-proxy executable is needed for ENABLE_BUBBLEWRAP_SANDBOX.  Unable to autodetect the path when cross-compiling. "
+                                "Please define define the CMake variable DBUS_PROXY_EXECUTABLE with the run-time full-path to the 'xdg-dbus-proxy' program.")
+        else ()
+            find_program(DBUS_PROXY_EXECUTABLE xdg-dbus-proxy)
+            if (NOT DBUS_PROXY_EXECUTABLE)
+                message(FATAL_ERROR "xdg-dbus-proxy executable not found and is needed for ENABLE_BUBBLEWRAP_SANDBOX. "
+                       "Either install it or use the CMake variable DBUS_PROXY_EXECUTABLE to define the runtime path.")
+            endif ()
         endif ()
-    elseif (NOT SILENCE_CROSS_COMPILATION_NOTICES)
-        message(NOTICE
-            "***--------------------------------------------------------***\n"
-            "***  Cannot check Bubblewrap version when cross-compiling. ***\n"
-            "***  The target system MUST have version 0.3.1 or newer.   ***\n"
-            "***  Use the BWRAP_EXECUTABLE and DBUS_PROXY_EXECUTABLE    ***\n"
-            "***  variables to set the run-time paths for the 'bwrap'   ***\n"
-            "***  and 'xdg-dbus-proxy' programs.                        ***\n"
-            "***--------------------------------------------------------***"
-        )
+    endif ()
+
+    # Do some extra sanity checks because the value of this paths has security implications if we get them wrong.
+    if (NOT "${BWRAP_EXECUTABLE}" MATCHES "^/.*/bwrap$")
```

There is an `IS_ABSOLUTE` conditional that can be used instead of regexes (faster, and clearer to understand):

```suggestion
    if (NOT IS_ABSOLUTE "${BWRAP_EXECUTABLE}")
```

I think it's not a good idea to check that the basename of the executable is `bwrap`. A distributor may install the program under a different name e.g. `bwrap-0.3` if they need different versions installed in parallel.

---

It looks to me like a really hypothetical situation, but in that case they could workaround that by shipping the binaries in different directories like `/usr/local/bwrap-0.3/bin/bwrap` which is also the easiest way of doing it (the package build scripts allow one to easily install in a different path prefix but not to change the binary name, that would have to be done post-install manually)

I find more likely to happen that someone by mistake passes a wrong path like `/usr/bin/bwra` and then nothing works at run-time